examples/research_projects/rag/lightning_base.py

```diff
@@ -167,8 +167,8 @@ def total_steps(self) -> int:
         effective_batch_size = self.hparams.train_batch_size * self.hparams.accumulate_grad_batches * num_devices
         return (self.dataset_size / effective_batch_size) * self.hparams.max_epochs
 
-    def setup(self, mode):
-        if mode == "test":
+    def setup(self, stage):
```

Comment:
why rename it here? renaming function inputs can cause backward breaking compability which should be avoided as much as possible. I don't see the benefit of renaming the arg here

Reply:
I had to change this according to the laters PL version.  Here, we are overriding the **setup** function and in the latest version they use "stage", instead of "mode". 

https://github.com/PyTorchLightning/pytorch-lightning/issues/7561#issuecomment-841690326